system/security/shared/caching.hpp

```diff
@@ -235,9 +239,11 @@ class CPermissionsCache : public CInterface
     map<string, ISecResource*>  m_managedFileScopesMap;
     mutable ReadWriteLock       m_scopesRWLock;//guards m_managedFileScopesMap
     ISecManager *               m_secMgr;
-    time_t                      m_lastManagedFileScopesRefresh;
 
     bool m_useLegacyDefaultFileScopePermissionCache = false;
+    Semaphore m_fileScopeCacheFillSem;
+    std::thread m_fileScopeCacheFillThread;
```

Comment:
question: Why use std::thread instead of CThread?  What advantages and disadvantages are there?